addon/run-task.ts

```diff
@@ -175,29 +175,36 @@ export function scheduleTask(
    @method throttleTask
    @param { Object } obj the instance to register the task for
    @param { String } taskName the name of the task to throttle
-   @param { Number } [timeout] the time in the future to run the task
+   @param { ...* } [throttleArgs] arguments to pass to the throttled method
+   @param { Number } timeout the time in the future to run the task
    @public
    */
 export function throttleTask(
   obj: IDestroyable,
-  taskName: any,
-  timeout: number = 0
+  taskName: string,
+  ...throttleArgs: any[]
 ): EmberRunTimer {
+
   assert(
     `Called \`throttleTask\` without a string as the first argument on ${obj}.`,
     typeof taskName === 'string'
   );
   assert(
-    `Called \`throttleTask('${taskName}', ${timeout})\` where '${taskName}' is not a function.`,
+    `Called \`throttleTask('${taskName}')\` where '${taskName}' is not a function.`,
     typeof obj[taskName] === 'function'
   );
   assert(
     `Called \`throttleTask\` on destroyed object: ${obj}.`,
     !obj.isDestroyed
   );
+  const timeout = throttleArgs[throttleArgs.length - 1];
```

Comment:
The param that `throttle` expects as the `number` is called `spacing`, which indicates the interval when the function will be invoked. Can we keep this consistent by renaming `timeout` to `spacing`?

Additionally, the last optional argument you can provide to `throttle` is `immediate`. Are we comfortable ignoring this param, or do you think we need an extra check to determine if we have a `number` or a `boolean`, and proceed accordingly?